pkg/gds/models/v1/models.go

```diff
@@ -173,6 +176,69 @@ func AppendCertReqID(vasp *pb.VASP, certreqID string) (err error) {
 	return nil
 }
 
+// NewCertificateRequest creates and returns a new certificate request to be associated with a VASP.
+func NewCertificateRequest(vasp *pb.VASP) (certRequest *CertificateRequest, err error) {
+	var (
+		organizationName    string
+		localityName        string
+		stateOrProvinceName string
+		countryName         string
+	)
+
+	if vasp == nil {
+		return nil, errors.New("must supply a VASP object for certificate request creation")
+	}
+
+	certRequest = &CertificateRequest{
+		Id:         uuid.New().String(),
+		Vasp:       vasp.Id,
+		CommonName: vasp.CommonName,
+		Params:     make(map[string]string),
+	}
+
+	// Populate the organization name, if available.
+	if vasp.Entity.Name != nil {
+		for _, name := range vasp.Entity.Name.NameIdentifiers {
+			if name.LegalPersonNameIdentifierType == ivms101.LegalPersonLegal {
+				organizationName = name.LegalPersonName
+				break
+			}
+		}
+	}
+	if organizationName != "" {
+		certRequest.Params["organizationName"] = organizationName
+	} else {
+		log.Info().
+			Str("vasp_id", vasp.Id).
+			Str("certreq_id", certRequest.Id).
+			Msg("organization name not found, populating new certificate request with default value")
+		certRequest.Params["organizationName"] = "TRISA Production"
+	}
+
+	// Populate the location information, if available.
+	if len(vasp.Entity.GeographicAddresses) > 0 {
+		address := vasp.Entity.GeographicAddresses[0]
+		localityName = address.TownLocationName
+		stateOrProvinceName = address.CountrySubDivision
+		countryName = address.Country
+	}
+	if localityName != "" && stateOrProvinceName != "" && countryName != "" {
+		certRequest.Params["localityName"] = localityName
+		certRequest.Params["stateOrProvinceName"] = stateOrProvinceName
+		certRequest.Params["countryName"] = countryName
```

Comment:
Should we allow partial location info? I don't think the TRISA address validation covers that case for us.

Reply:
Oh man, this is going to be a pain in the butt. You're right - we should not have partial location info, and right now the front end uses `AddressLines` instead of `TownLocationName` and `CountrySubDivision` … You'll definitely have `countryName` but you won't have TownLocationName and CountrySubDivision. I don't want `Menlo Park, California, SG`, so I think the check you're doing right now is right.

Let's create a follow on story to see if locality and stateorprovince are optional in Sectigo. If so, we can move forward with just country. We'll then also probably have to create follow on stories to allow these fields to be edited from the Admin UI.